src/Features/Core/Portable/QuickInfo/IndentationHelper.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    // Reproduces logic in IProjectionBufferFactoryServiceExtensions (editor layer) 
```

Comment:
Mostly the same logic, but is dealing in different terms, SourceText instead of ITextSnapshot, and results in creating positive text ranges instead of elision ranges.

---
In reply to: [77691837](https://github.com/dotnet/roslyn/pull/13623#discussion_r77691837) [](ancestors = 77691837)